tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/OpenTelemetrySpan.java

```diff
@@ -100,7 +107,12 @@ public Scope activate() {
     public Span baggage(String key, String value) {
         Objects.requireNonNull(key, "baggage key cannot be null");
         Objects.requireNonNull(value, "baggage value cannot be null");
-        baggage.baggage(key, value);
+        if (baggage instanceof MutableOpenTelemetryBaggage mutableBaggage) {
```

Comment:
Is there maybe a way to switch to mutable baggage here and make it work, or is this a limitation of the open telemetry API?
If we have this check + exception, then an application that works fine with other tracing provider may fail with oTel